docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+                history = q.get()
+                p.join()
+                history_dict = history.history
+                conf = self.provider.full_config
+                self.info("Setting new history")
+
+                # Pot, meet kettle. Need another method or opportunity for Additional Execercise section?
+                ui_history = {}
+
+                ui_history["trainingLoss"] = history_dict["loss"]
+                ui_history["trainingBinaryAccuracy"] = history_dict["binary_accuracy"]
+                ui_history["validationLoss"] = history_dict["val_loss"]
+                ui_history["validationBinaryAccuracy"] = history_dict["val_binary_accuracy"]
+                conf['Configuration']["modelEvaluation"]['history'] = ui_history
+                self.provider.save_full_config(conf)
+            elif self.MODE == 'EVAL':
+                model = tf.keras.saving.load_model(self.provider.tool_config["modelConfig"]["modelName"], custom_objects={'custom_standardization': custom_standardization})
+            elif self.MODE == 'PREVIEW':
+                self.send_preview_data()
+```
+
+
+The majority of the above speaks for itself at this point in the guide.
+Our most notable (conceptual) addition is likely the `MultiProcessing` section:  
+
+```python
+q = Queue()
+p = Process(target=self.create_and_save_model, args=(q, *fn_args))
+p.start()
+history = q.get()
+p.join()
+```
+
+Note that this is "just enough" to prevent blocking. An incredible amount of resources are available online and on Python's official documentation. For our purposes, know we're using the `Process(...)` to wrap up our expensive compute task - allowing us to avoid blocking the Python server IO.
+
+Now the backend can generate, train, and deploy/export a new model! All that's left now is our final step, allowing the User to use the "production" model to predict data in a workflow!
+
+
+### `PREDICT`
+
+In our last mode, we must update our `on_record_batch` function to call `model.predict(...)` on values passed to us in the workflow. In our case, we will eventually use a text input for testing.
+**NOTE** However, this could be _any_ sort of record batch, as long as it can connect to our anchor and with the appropriate code updates!
+In other words, allow Users to use our "production" version for predictions.
+
+Some examples of why one might do this include:
+* a tool developer can ship a trained model that exemplifies expected behavior alongside "starter" training data for new user-generated and trained models.
+* A tool developer may ship a "base" model, which the User may then tune and train further in their workflow.
+* With additional UI, users may choose how/where/when to deploy models in other workflows or jobs.
+* Workflows running on Server may do **any of the above to generate serious, performant, and production-grade models**, which also
+* Empowers tool **developers AND users** to create production-grade TensorFlow AI/ML pipelines!
```

Comment:
TW review:

Some examples of why this might be done:
* A tool developer can ship a trained model that exemplifies expected behavior alongside starter training data for new user-generated and trained models.
* A tool developer might ship a base model, which the user might then tune and train further in their workflow.
* With additional UI, users might choose how, where, and when to deploy models in other workflows or jobs.
* Workflows that run on Server might do _any of the above to generate serious, performant, and production-grade models_, which also...
* Empowers tool developers _and_ users to create production-grade TensorFlow AI/ML pipelines!